ccaas_builder/cmd/build/main.go

```diff
@@ -0,0 +1,175 @@
+	if err != nil {
+		return err
+	}
+
+	var metadata ChaincodeMetadata
+	if err := json.Unmarshal(metadataFileContents, &metadata); err != nil {
+		return err
+	}
+
+	if strings.ToLower(metadata.Type) != "ccaas" {
+		return fmt.Errorf("chaincode type should be ccaas, it is %s", metadata.Type)
+	}
+
+	if err := copy.Copy(metadataDir, outputDir); err != nil {
+		return fmt.Errorf("failed to copy build metadata folder: %s", err)
+	}
+
+	if _, err := os.Stat(metainfoSrcDir); !os.IsNotExist(err) {
+		if err := copy.Copy(metainfoSrcDir, metainfoDestDir); err != nil {
+			return fmt.Errorf("failed to copy build META-INF folder: %s", err)
+		}
+	}
+
+	// Process and update the connections file
+	fileInfo, err := os.Stat(connectionSrcFile)
+	if err != nil {
+		return fmt.Errorf("connection.json not found in source folder: %s", err)
+	}
+
+	connectionFileContents, err := ioutil.ReadFile(connectionSrcFile)
+	if err != nil {
+		return err
+	}
+
+	// read the connection.json file into structure to process
+	var connectionData Connection
+	if err := json.Unmarshal(connectionFileContents, &connectionData); err != nil {
+		return err
+	}
+
+	// Treat each of the string fields in the connection.json as Go template
+	// strings. They can be fixed strings, but if they are templates
+	// then the JSON string that is defined in CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG
+	// is used as the 'context' to parse the string
+
+	updatedConnection := Connection{}
+	var cfg map[string]interface{}
+
+	cfgString := os.Getenv("CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG")
+	if cfgString != "" {
+		if err := json.Unmarshal([]byte(cfgString), &cfg); err != nil {
+			return fmt.Errorf("Failed to unmarshal %s", err)
+		}
+	}
+
+	updatedConnection.Address, err = execTempl(cfg, connectionData.Address)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the Address field template: %s", err)
+	}
+
+	updatedConnection.DialTimeout, err = execTempl(cfg, connectionData.DialTimeout)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the DialTimeout field template: %s", err)
+	}
+
+	if connectionData.TLS {
+		updatedConnection.TLS = true
+		updatedConnection.ClientAuth = connectionData.ClientAuth
+
+		updatedConnection.RootCert, err = execTempl(cfg, connectionData.RootCert)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the RootCert field template: %s", err)
+		}
+		updatedConnection.ClientKey, err = execTempl(cfg, connectionData.ClientKey)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the ClientKey field template: %s", err)
+		}
+
+	} else {
+		updatedConnection.TLS = false
+	}
+
+	updatedConnectionBytes, err := json.Marshal(updatedConnection)
+	if err != nil {
+		return fmt.Errorf("failed to marshal updated connection.json file: %s", err)
+	}
+
+	err = ioutil.WriteFile(connectionDestFile, updatedConnectionBytes, fileInfo.Mode())
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// Helper function to process a template against a string, and return a string
```

Comment:
function comments in Go start with the function name, e.g.
// execTempl is a helper function to...